[StaticRuntime] Use NNC's call_raw API to reduce call overheads.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57329 [StaticRuntime] Use NNC's call_raw API to reduce call overheads.**
* #57328 [TensorExpr] Add `TensorExprKernel::runFast` method.
* #55113 [TensorExpr] Add `CodeGen::call_raw` method.

Differential Revision: [D28110358](https://our.internmc.facebook.com/intern/diff/D28110358)

Commit:
[StaticRuntime] Use NNC's call_raw API to reduce call overheads.

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/runtime/static/ops.cpp:
```diff
@@ -379,6 +379,10 @@ struct TEWrapper {
     cg->call(args);
   }
```
Can we rip out the `operator()` overload if we're not going to use it anymore?

ZolotukhinM replied:
We probably can

ZolotukhinM replied:
Done!

bertmaher reviewed (approved):
wicked!

Commit:
Update on "[StaticRuntime] Use NNC's call_raw API to reduce call overheads."


Differential Revision: [D28110358](https://our.internmc.facebook.com/intern/diff/D28110358)

[ghstack-poisoned]

mruberry commented:
Unlanding this stack, as it appears to be breaking pytorch_linux_xenial_py3_clang5_asan_test2. Failure snippet:

```
Apr 30 20:18:38 [ RUN      ] Kernel.RunFast
Apr 30 20:18:38 unknown file: Failure
Apr 30 20:18:38 C++ exception with description "SimpleIREvaluator::call_raw is not implemented yet" thrown in the test body.
Apr 30 20:18:38 [  FAILED  ] Kernel.RunFast (7 ms)
```

The failure appears in the PR CI
